designer/server/src/main/scala/pl/touk/nussknacker/ui/process/test/ScenarioTestDataSerDe.scala

```diff
@@ -1,47 +1,69 @@
 package pl.touk.nussknacker.ui.process.test
 
-import pl.touk.nussknacker.engine.api.test.{TestData, TestRecord}
+import io.circe.syntax.EncoderOps
+import io.circe.{Decoder, Encoder, Json, parser}
+import pl.touk.nussknacker.engine.api.test.{ScenarioTestData, ScenarioTestRecord}
+import pl.touk.nussknacker.ui.api.TestDataSettings
 
 import java.nio.charset.StandardCharsets
 import scala.io.Source
 import scala.util.Using
-import io.circe.parser
-import pl.touk.nussknacker.ui.api.TestDataSettings
 
-// TODO multiple-sources-test: switch TestData to ScenarioTestData
+object ScenarioTestDataSerDe {
+
+  private implicit val scenarioTestRecordEncoder: Encoder[ScenarioTestRecord] = Encoder.instance(scenarioTestRecord =>
+    Json.obj(
+      "sourceId" -> Json.fromString(scenarioTestRecord.sourceId.id),
+      "record" -> scenarioTestRecord.record.json,
+      "timestamp" -> scenarioTestRecord.record.timestamp.asJson
+    ).dropNullValues
+  )
+
+  private implicit val scenarioTestRecordDecoder: Decoder[ScenarioTestRecord] = Decoder.instance(hcursor =>
+    for {
+      sourceId <- hcursor.downField("sourceId").as[String]
+      record <- hcursor.downField("record").as[Json]
+      timestamp <- hcursor.downField("timestamp").as[Option[Long]]
+    } yield ScenarioTestRecord(sourceId, record, timestamp)
+  )
+
+}
+
 class ScenarioTestDataSerDe(testDataSettings: TestDataSettings) {
 
-  def serializeTestData(testData: TestData): Either[String, RawScenarioTestData] = {
-    val content = testData.testRecords
-      .map(_.json.noSpaces)
+  import ScenarioTestDataSerDe._
+
+  def serializeTestData(scenarioTestData: ScenarioTestData): Either[String, RawScenarioTestData] = {
```

Comment:
Switched to String since test data is actually lines of JSON so it's cannot be parsed easily by akka. However I also renamed `testDataMaxBytes` to `testDataMaxLength`.